embedded-graphics/src/primitives/polyline.rs

```diff
@@ -0,0 +1,358 @@
+            });
+
+        Rectangle::with_corners(top_left, bottom_right)
+    }
+}
+
+impl<'a> Transform for Polyline<'a> {
+    /// Translate the polyline from its current position to a new position by (x, y) pixels, returning
+    /// a new `Polyline`. For a mutating transform, see `translate_mut`.
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Polyline;
+    /// # use embedded_graphics::prelude::*;
+    /// let points = [
+    ///     Point::new(5, 10),
+    ///     Point::new(7, 7),
+    ///     Point::new(5, 8),
+    ///     Point::new(10, 10),
+    /// ];
+    ///
+    /// let polyline = Polyline::new(&points);
+    /// let moved = polyline.translate(Point::new(10, 12));
+    ///
+    /// assert_eq!(polyline.bounding_box().top_left, Point::new(5, 7));
+    /// assert_eq!(moved.bounding_box().top_left, Point::new(15, 19));
+    /// ```
+    fn translate(&self, by: Point) -> Self {
+        Self {
+            translate: by,
+            ..*self
+        }
+    }
+
+    /// Translate the polyline from its current position to a new position by (x, y) pixels.
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Polyline;
+    /// # use embedded_graphics::prelude::*;
+    /// let points = [
+    ///     Point::new(5, 10),
+    ///     Point::new(7, 7),
+    ///     Point::new(5, 8),
+    ///     Point::new(10, 10),
+    /// ];
+    ///
+    /// let mut polyline = Polyline::new(&points);
+    ///
+    /// polyline.translate_mut(Point::new(10, 12));
+    ///
+    /// assert_eq!(polyline.bounding_box().top_left, Point::new(15, 19));
+    /// ```
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.translate += by;
+
+        self
+    }
+}
+
+/// An iterator over all pixel positions on the polyline
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct PolylineIterator<'a> {
+    stop: bool,
+    vertices: &'a [Point],
+    translate: Point,
+    segment_iter: ThickLineIterator,
+}
+
+impl<'a> Iterator for PolylineIterator<'a> {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.stop {
+            return None;
+        }
+
+        if let Some(p) = self.segment_iter.next() {
+            Some(p)
+        } else {
+            let (start, rest) = self.vertices.split_first()?;
+            let end = rest.get(0)?;
+
+            self.vertices = rest;
+
+            self.segment_iter = ThickLineIterator::new(
+                &Line::new(*start + self.translate, *end + self.translate),
+                1,
+            );
+
+            // Skip first point of next line, otherwise we overlap with the previous line
+            Self::next(self);
+
+            Self::next(self)
+        }
+    }
+}
+
+impl<'a> IntoIterator for Polyline<'a> {
+    type Item = Point;
+    type IntoIter = PolylineIterator<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.vertices
```

Comment:
Could you always initialize `segment_iter` with an empty iterator and move all conditional code into `next`? IMO that might result in cleaner code.

Reply:
Great idea. I thought I was being all clever with those `.map()`s...

Reply:
I changed it, but changed it back (although with the `ThickLineIterator` -> `line::Points` change) as the first segment need to render its first point, but subsequent segments skip their first point in the `next()` impl. I couldn't think of a clean way to check for segment index in `next()` so left this approach here.